packages/blocks/src/store/selectors.js

```diff
@@ -22,203 +37,168 @@ import {
  * Given a block name or block type object, returns the corresponding
  * normalized block type object.
  *
- * @param {Object}          state      Blocks state.
  * @param {(string|Object)} nameOrType Block name or type object
  *
  * @return {Object} Block type object.
  */
-const getNormalizedBlockType = ( state, nameOrType ) =>
+const getNormalizedBlockType = ( nameOrType ) => ( { get } ) =>
 	'string' === typeof nameOrType
-		? getBlockType( state, nameOrType )
+		? get( blockTypesFamily( nameOrType ) )
 		: nameOrType;
 
 /**
  * Returns all the available block types.
  *
- * @param {Object} state Data state.
- *
  * @return {Array} Block Types.
  */
-export const getBlockTypes = createSelector(
-	( state ) => {
-		return Object.values( state.blockTypes ).map( ( blockType ) => {
-			return {
-				...blockType,
-				variations: getBlockVariations( state, blockType.name ),
-			};
-		} );
-	},
-	( state ) => [ state.blockTypes, state.blockVariations ]
-);
+export const getBlockTypes = () => ( { get } ) => {
+	return get( allBlockTypes );
+};
 
 /**
  * Returns a block type by name.
  *
- * @param {Object} state Data state.
  * @param {string} name Block type name.
  *
  * @return {Object?} Block Type.
  */
-export function getBlockType( state, name ) {
-	return state.blockTypes[ name ];
-}
+export const getBlockType = ( name ) => ( { get } ) => {
+	return get( blockTypesFamily( name ) );
+};
 
 /**
  * Returns block styles by block name.
  *
- * @param {Object} state Data state.
  * @param {string} name  Block type name.
  *
  * @return {Array?} Block Styles.
  */
-export function getBlockStyles( state, name ) {
-	return state.blockStyles[ name ];
-}
+export const getBlockStyles = ( name ) => ( { get } ) => {
+	return get( blockStylesFamily( name ) );
+};
 
 /**
  * Returns block variations by block name.
  *
- * @param {Object}                state     Data state.
  * @param {string}                blockName Block type name.
  * @param {WPBlockVariationScope} [scope]   Block variation scope name.
  *
  * @return {(WPBlockVariation[]|void)} Block variations.
  */
-export function getBlockVariations( state, blockName, scope ) {
-	const variations = state.blockVariations[ blockName ];
+export const getBlockVariations = ( blockName, scope ) => ( { get } ) => {
```

Comment:
While working on this selector and other similar selectors, I realized something important. Right now these selectors recompute on each mapSelectToProps but ideally they are built using `WPAtomFamily` technique to recompute only when their dependencies change.

The problem is that `WPAtomFamily` only accepts one argument for the "key" :) while selectors have an arbitrary number of arguments. So we need something like `WPAtomFamilySelector` here and if you think about it `WPAtomFamily` is just a specific case (a selector with only one argument). So I think the API would gain in clarity if we remove `WPAtomFamily` and implement a more generic ` WPAtomSelector`. I'll explore that separately.

Reply:
The selector issue is being solved here https://github.com/WordPress/gutenberg/pull/27147